src/cts/src/TritonCTS.cpp

```diff
@@ -517,28 +523,152 @@ std::string TritonCTS::selectRootBuffer(std::vector<std::string>& bufferVector)
     bufferVector = options_->getBufferList();
   }
 
-  // pick the lowest driver resistance as root
-  float bestRes = std::numeric_limits<float>::max();
-  std::string rootBuf;
+  if (bufferVector.size() == 1) {
+    return bufferVector.front();
+  }
+
+  // estimate wire cap for root buffer
+  // assume root buffer needs to drive clk buffers at two far ends of chip
+  //
+  //  --------------
+  //  |      .     |
+  //  |      .     |
+  //  |===== x ====|
+  //  |      .     |
+  //  |      .     |
+  //  --------------
+  odb::dbBlock* block = db_->getChip()->getBlock();
+  odb::Rect coreArea = block->getCoreArea();
+  int width = coreArea.xMax() - coreArea.xMin();
+  int height = coreArea.yMax() - coreArea.yMin();
+  float rootWireLength = static_cast<float>(std::max(width, height))
+                         / block->getDbUnitsPerMicron();
+  sta::Corner* corner = openSta_->cmdCorner();
+  float rootWireCap
+      = resizer_->wireClkCapacitance(corner) * 1e-6 * rootWireLength;
+  // clang-format off
+  debugPrint(logger_, CTS, "buffering", 2, "core width:{}, core height:{}, "
+             "rootWL:{:0.2e}, rootWireCap:{:0.2e}, clock cap/micron:{:0.2e}", width, height,
+             rootWireLength, rootWireCap,
+             resizer_->wireClkCapacitance(corner) * 1e-6);
+  // clang-format on
+
+  // pick the smallest buffer that can drive root wire cap
+  // if no such buffer exists, pick one that has the largest max cap
+  float bestArea = std::numeric_limits<float>::max();
```

Comment:
This duplicated in TritonCTS::selectSinkBuffer.  Please make one shared copy.

Reply:
Created selectBestMaxCapBuffer and shared it with two routines